rust/crd/src/lib.rs

```diff
@@ -78,43 +83,63 @@ pub enum Error {
 )]
 #[serde(rename_all = "camelCase")]
 pub struct HdfsClusterSpec {
+    /// Configuration that applies to all roles and role groups.
+    /// This includes settings for authentication, logging and the ZooKeeper cluster to use.
+    pub cluster_config: HdfsClusterConfig,
+
+    // no doc string - See ProductImage struct
     pub image: ProductImage,
+
+    // no doc string - See ClusterOperation struct
+    #[serde(default)]
+    pub cluster_operation: ClusterOperation,
+
+    // no doc string - See Role struct
     #[serde(default, skip_serializing_if = "Option::is_none")]
     pub name_nodes: Option<Role<NameNodeConfigFragment>>,
+
+    // no doc string - See Role struct
     #[serde(default, skip_serializing_if = "Option::is_none")]
     pub data_nodes: Option<Role<DataNodeConfigFragment>>,
+
+    // no doc string - See Role struct
     #[serde(default, skip_serializing_if = "Option::is_none")]
     pub journal_nodes: Option<Role<JournalNodeConfigFragment>>,
-    // Cluster wide configuration
-    pub cluster_config: HdfsClusterConfig,
-    /// Cluster operations like pause reconciliation or cluster stop.
-    #[serde(default)]
-    pub cluster_operation: ClusterOperation,
 }
 
 #[derive(Clone, Debug, Deserialize, Eq, Hash, JsonSchema, PartialEq, Serialize)]
 #[serde(rename_all = "camelCase")]
 pub struct HdfsClusterConfig {
+    /// `dfsReplication` is the factor of how many times a file will be replicated to differen data nodes.
+    /// The default is 3.
+    /// You need at this many data nodes so each file can be replicated correctly, otherwise a warning will be printed.
```

Comment:
```suggestion
    /// You need at least the same amount of data nodes so each file can be replicated correctly, otherwise a warning will be printed.
```